src/state.cairo

```diff
@@ -46,22 +50,34 @@ pub struct Header {
     pub nonce: u32,
 }
 
-/// Transaction
-/// https://developer.bitcoin.org/reference/transactions.html#raw-transaction-format
+/// Extended transaction.
+/// https://learnmeabitcoin.com/technical/transaction/
+///
+/// Contains additional "meta" fields required for validation.
 #[derive(Drop, Copy)]
 pub struct Transaction {
```

Comment:
aren't we missing `input_count` and `output_count` fields ?

Reply:
They can be obtained from inputs/outputs spans so it probably does not make much sense to specify them explicitly